app/main.py

```diff
@@ -2,0 +1,34 @@
+import datetime
+from typing import List, Dict, Any
+from app.cafe import Cafe
+
+
+def go_to_cafe(friends: List[Dict[str, Any]], cafe: Cafe) -> str:
+    """
+    :param friends: A list of friends,
+     each represented by a dictionary with their details.
+    :param cafe: An instance of the Cafe class
+     representing the cafe being visited.
+    :return: A message about the visit to the cafe.
+    """
+    for friend in friends:
+        if "vaccine" not in friend:
+            return "All friends should be vaccinated"
+
+    not_vaccinated_count = sum(
+        1 for friend in friends
+        if friend["vaccine"]["expiration_date"] < datetime.date.today())
+    mask_needed_count = sum(
+        1 for friend in friends
+        if not friend.get("wearing_a_mask", True))
+
+    if not_vaccinated_count > 0:
+        return "All friends should be vaccinated"
+
+    if mask_needed_count > 0:
+        if mask_needed_count == 1:
+            return f"Friends should buy {mask_needed_count} mask"
+        else:
+            return f"Friends should buy {mask_needed_count} masks"
```

Comment:
Code Style: There is no need to use `else` after `return` in the `if` block. When the `if` condition is `True`, the function will return and won't execute the rest of the code. If the condition is `False`, the function will continue executing the next line of code, which is the line you've put in the `else` block. So you can remove `else` here.